birdhouse/read-configs.include.sh

```diff
@@ -352,20 +352,20 @@ process_backwards_compatible_variables() {
 
       if [ x"${BIRDHOUSE_BACKWARD_COMPATIBLE_ALLOWED}" = x"True" ] && [ ! "$1" = "pre-components" ]; then
         # Enable corresponding old var for template expansion if new var is in template expansion.
-        if echo "${VARS}" | grep -q "\s\$${new_var}\b"; then
+        if echo "${VARS}" | grep -q "[[:space:]]\$${new_var}[[:space:]]*"; then
```

Comment:
This is better but you're going to want to be able to handle VARS that start or end at the edge of the line as well (see the suggestion here: https://github.com/bird-house/birdhouse-deploy/pull/518#discussion_r2052929257)

Right now for example, if `new_var='$SOME_VAR'`, this won't match:

```sh
VARS='$SOME_VAR'
VARS='
$SOME_VAR'
```

and it will match:

```sh
VARS=' $SOME_VAR_OTHER_THING_ENTIRELY'
VARS='
  $SOME_VAR_ANOTHER_ONE
'
```

both of which you don't want.

Reply:
> if `new_var='$SOME_VAR'`, this won't match:

`VARS` is an append only variable so setting it directly like that should never happen.  Not only the user will zap their own variable, they will zap all the variables previously accumulated there !



> ```shell
> VARS=' $SOME_VAR_OTHER_THING_ENTIRELY'
> ```

This is why I previously had `\b` in `${new_var}\b` which will match the edge of the word and would take care of this scenario, but then you said it is not portable.

Each append to `VARS` is on a new line is just a readability thing.  It's never enforced or even documented.  Anyone can do `VARS="$VARS \$NEW_1 \$NEW_2 \$NEW_3"` and it is still valid.

Unfortunately I have not found `\b` portable equivalent.  So to handle this case we can use `\b`?

Reply:
Or you can do:

```sh
echo ${VARS} | grep -q '\(^\|[[:space:]]\)\$${new_var}\([[:space:]]\|$\)'
```

Or:

```sh
echo "|$(echo ${VARS}| tr ' ' \|)|" | grep "|\$${new_var}|"
```

Or:

```sh
echo " ${VARS} " | tr '\n' ' ' | grep "[[:space:]]\$${new_var}[[:space:]]"
```

Reply:
> ```shell
> echo " ${VARS} " | tr '\n' ' ' | grep "[[:space:]]\$${new_var}[[:space:]]"
> ```

Ah nice, adding extra "space" padding to normalize everything to space !  Why haven't I though about that!  I like this last one, it's the simplest of all the other variations.  

The `|` padded version might not handle the case when 
```
VARS="$VARS
\$NEW"  # no space in front to turn to '|'
```

The other 2 with or `\|` operator and meta `\(` and `\)` just seems too easy to make a typo mistake and takes more time to "read" and "understand".

Reply:
All of them work. I'll leave it up to you which you want to use.